awx/main/tasks.py

```diff
@@ -407,7 +405,26 @@ def cleanup_execution_environment_images():
 
 @task(queue=get_local_queuename)
 def check_heartbeat(node):
-    AWXReceptorJob.check_heartbeat(node)
+    instance = Instance.objects.get(hostname=node)
+    data = worker_info(node)
+
+    if data['Errors']:
+        formatted_error = "\n".join(data["Errors"])
+        logger.warn(f'Failed to find capacity of execution node {node}, errors:\n{formatted_error}')
+    else:
+        # TODO: spin off new instance method from refresh_capacity that calculates derived fields
+        instance.cpu = data['CPU Capacity']  # TODO: rename field on runner side to not say "Capacity"
+        instance.cpu_capacity = get_cpu_effective_capacity(instance.cpu)
```

Comment:
@beeankha see how we have the split `cpu` / `cpu_capacity` fields? What we want from runner is "raw" units, like "number of CPUs", so because of that, I think "CPU Capacity" is not a good name. I'd like to go through a rename, and also dump the capital first letters.